crates/database/src/repositories/file.rs

```diff
@@ -21,41 +22,43 @@ impl FileRepository {
         &self,
         params: services::files::ports::CreateFileParams,
     ) -> Result<File, RepositoryError> {
-        let client = self
-            .pool
-            .get()
-            .await
-            .context("Failed to get database connection")
-            .map_err(RepositoryError::PoolError)?;
-
         let id = Uuid::new_v4();
         let now = Utc::now();
 
-        let row = client
-            .query_one(
-                r#"
+        let row = retry_db!("create_new_file_record", {
+            let client = self
+                .pool
+                .get()
+                .await
+                .context("Failed to get database connection")
+                .map_err(RepositoryError::PoolError)?;
+
+            client
+                .query_one(
+                    r#"
                 INSERT INTO files (
```

Comment:
We should handle the case where:
```
INSERT succeeds → connection drops → retry attempts INSERT with same UUID → UNIQUE_VIOLATION error → operation reported as failed even though record exists
```

INSERT ... ON CONFLICT should be able to fix this issue.